src/main/java/net/snowflake/ingest/streaming/internal/ParquetFlusher.java

```diff
@@ -112,6 +115,23 @@ private SerializationResult serializeFromParquetWriteBuffers(
       }
 
       rowCount += data.getRowCount();
+
+      // We check if the number of rows collectively written for channels encountered so far matches
+      // the number of rows in metadata
+      if (mergedChannelWriter.getRowsWritten() != rowCount) {
```

Comment:
Can this be potentially racy i.e. other threads writing data via the writer concurrently? Or is the `mergedChannelWriter` only used from this current thread?